File: src-tauri/src/models/metadata.rs
#[derive(Debug, serde::Serialize, serde::Deserialize, sqlx::FromRow)]
pub struct ImageMetadata {
    pub asset_id: i64,
    pub width: Option<i32>,
    pub height: Option<i32>,
    pub color_profile: Option<String>,
    pub dpi: Option<f64>,
    /// Stored as 0/1 in SQLite
    pub has_alpha: bool,
}

#[derive(Debug, serde::Serialize, serde::Deserialize, sqlx::FromRow)]
pub struct VideoMetadata {
    pub asset_id: i64,
    pub width: Option<i32>,
    pub height: Option<i32>,
    pub duration: Option<f64>,
    pub frame_rate: Option<f64>,
    pub codec: Option<String>,
    pub audio_codec: Option<String>,
    pub bit_rate: Option<i64>,
}

File: src/types/metadata.ts
// Rust serializes struct fields as snake_case — TypeScript types match.

import type { Asset } from "./asset";
import type { Tag } from "./tag";

export interface ImageMetadata {
  asset_id: number;
  width: number | null;
  height: number | null;
  color_profile: string | null;
  dpi: number | null;
  has_alpha: boolean;
}

export interface VideoMetadata {
  asset_id: number;
  width: number | null;
  height: number | null;
  duration: number | null;
  frame_rate: number | null;
  codec: string | null;
  audio_codec: string | null;
  bit_rate: number | null;
}

/** Full asset detail returned by get_asset_detail — asset fields flattened + metadata + tags. */
export interface AssetDetail extends Asset {
  image_metadata: ImageMetadata | null;
  video_metadata: VideoMetadata | null;
  tags: Tag[];
}
